src/realm/group_shared.cpp

```diff
@@ -1320,7 +1320,7 @@ void SharedGroup::do_open(const std::string& path, bool no_create_file, bool is_
 
 // WARNING / FIXME: compact() should NOT be exposed publicly on Windows because it's not crash safe! It may
 // corrupt your database if something fails
-bool SharedGroup::compact(bool bump_version_number)
+bool SharedGroup::compact(bool bump_version_number, util::Optional<const char*> output_encryption_key)
```

Comment:
Do you really want to wrap the pointer in `util::Optional<>`? it just seems very heavy handed to me :-)

Reply:
I realised that we need a way to differentiate from using the existing key (if any), to changing to an unencrypted file (nullptr). This means there is different functionality when passing `none` and `nullptr`.

Reply:
Oh, you want to be able to go from encrypted to unencrypted. Better explain those different meanings of `none` and `null`.